tests/test_authorisation.py

```diff
@@ -351,9 +340,136 @@ def test_different_user_levels_have_correct_flagging_permissions(
                     "assess/flag/app123",
                     follow_redirects=True,
                 )
+                assert False, "Did not throw expected error."
             except Exception as e:
                 # Redirect to authenticator
                 assert (
                     e.args[0]
                     == "Following external redirects is not supported."
                 )
+
+    @pytest.mark.parametrize(
+        "claim,expect_resolve_flag_available",
+        [
+            (test_commenter_claims, False),
+            (test_assessor_claims, False),
+            (test_lead_assessor_claims, True),
+        ],
+    )
+    def test_different_user_levels_have_correct_permissions_to_resolve_flag(
+        self,
+        flask_test_client,
+        claim,
+        expect_resolve_flag_available,
+        mock_get_banner_state,
+    ):
+        """
+        GIVEN authorised users
+        WHEN the user accesses the resolve flag route
+        THEN the user sees the appropriate page:
+            - commentor role cannot resolve flag
+            - assessor role cannot resolve flag
+            - lead assessor role can resolve flag
+        """
+
+        flask_test_client.set_cookie(
+            "localhost",
+            "fsd_user_token",
+            create_valid_token(claim),
+        )
+
+        if expect_resolve_flag_available:
+            response = flask_test_client.get(
+                "assess/resolve_flag/app123",
+                follow_redirects=True,
+            )
+            assert response.status_code == 200
+            assert b"Resolve flag" in response.data
+            assert b"Query resolved" in response.data
+            assert b"Stop assessment" in response.data
+        else:
+            try:
+                flask_test_client.get(
+                    "assess/resolve_flag/app123",
+                    follow_redirects=True,
+                )
+                assert False, "Did not throw expected error."
+            except Exception as e:
+                # Redirect to authenticator
+                assert (
+                    e.args[0]
+                    == "Following external redirects is not supported."
+                )
```

Comment:
If you end up keeping this logic, and not splitting them into two separate tests, but you want to use pytest.rasises, - you can parametrize the context manager and use contextlib.nullcontext() for exception scenarios (let me know if you want me to elaborate on this, or give some examples)

Reply:
ill give your suggestion a go, looks cleaner and more intuitive 👍